packages/react/src/useAction.ts

```diff
@@ -75,7 +75,8 @@ export const useAction = <
   return [
     transformedResult,
     useCallback(
-      async (variables, context) => {
+      async (variables: F["variablesType"], context?: Partial<OperationContext>) => {
```

Comment:
I think I freaked out typescript too much by changing the return type of this outer function to require an inference, this just fixes internal type errors within these functions but doesn't change the output type